EKrut-Server/src/ekrut/server/managers/ServerInventoryManager.java

```diff
@@ -0,0 +1,86 @@
+package ekrut.server.managers;
+
+import ekrut.server.db.InventoryItemDAO;
+import ekrut.entity.InventoryItem;
+import ekrut.net.InventoryItemRequest;
+import ekrut.net.InventoryItemResponse;
+
+public class ServerInventoryManager {
+
+	// pub-sending notifications
+	public InventoryItemResponse updateItemQuantity(InventoryItemRequest inventoryUpdateItemRequest) {
+		if (inventoryUpdateItemRequest == null)
+			throw new IllegalArgumentException("null InventoryItemRequest was provided.");
+		
+		// Unpack inventoryUpdateItemRequest into it's components.
+		int itemId = inventoryUpdateItemRequest.getItemId();
+		int quantity = inventoryUpdateItemRequest.getQuantity();
+		String ekrutLocation = inventoryUpdateItemRequest.getEkrutLocation();
+		
+		// Check that updated quantity is a valid value.
+		if (quantity < 0)
+			return new InventoryItemResponse("Updated quantity must be a non-negative integer.");
+		
+		// Fetch InventoryItem from DB.
+		InventoryItem inventoryItemInDB = InventoryItemDAO.fetchInventoryItem(itemId, ekrutLocation);
+		
+		// Check that InventoryItem exist in DB.
+		if (inventoryItemInDB == null)
+			return new InventoryItemResponse("Couldn't find InventoryItem in DB.");
+		
+		//Check if new quantity is below the threshold of that InventoryItem.
+		if (quantity < inventoryItemInDB.getItemThreshold())
+			// In order to send 'below threshold' notification we need access to UserNotifier &
+			// To know the Area manager's information (who is the manager of this machine??)
+			// TBD HOW TO SEND NOTIFICATIONS??
```

Comment:
Comments are not statements. If you want to leave this if statement empty for now you need to do
```java
if (cond) {}
```
As it is right now, the following if statement is actually nested inside of this if.
This:
```java
if (quantity < inventoryItemInDB.getItemThreshold())
	// In order to send 'below threshold' notification we need access to UserNotifier &
	// To know the Area manager's information (who is the manager of this machine??)
	// TBD HOW TO SEND NOTIFICATIONS??

// Try to commit the update in DB.
if (!InventoryItemDAO.updateItemQuantity(itemId,quantity, ekrutLocation))
	return new InventoryItemResponse("Failed updating item quantity.");
```
is equivalent to this:
```java
if (quantity < inventoryItemInDB.getItemThreshold())
        if (!InventoryItemDAO.updateItemQuantity(itemId,quantity, ekrutLocation))
	        return new InventoryItemResponse("Failed updating item quantity.");
```